### PR TITLE
[ADF-4738] [Process -Cloud] Move GroupCloudService to adf-core

### PR DESCRIPTION
**Pull request by sivakumar414ram**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

    https://issues.alfresco.com/jira/browse/ADF-4738

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/userinfo/services/identity-group.service.ts [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+                        }
+                    });
+                }
+                return hasRole;
+            }),
+            catchError((error) => this.handleError(error))
+        );
+    }
+
+    private groupClientRoleMappingApi(groupId: string, clientId: string): any {
```
Comment (DenysVuika) [inline lib/core/userinfo/services/identity-group.service.ts]: missing return type
```suggestion
    private groupClientRoleMappingApi(groupId: string, clientId: string): string {
```

File: lib/core/userinfo/services/identity-group.service.ts [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+        return `${this.appConfigService.get('identityHost')}/groups/${groupId}/role-mappings/clients/${clientId}`;
+    }
+
+    private getApplicationIdApi() {
```
Comment (DenysVuika) [inline lib/core/userinfo/services/identity-group.service.ts]: Missing return type
```suggestion
    private getApplicationIdApi(): string {
```

File: lib/core/userinfo/services/identity-group.service.ts [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+        return `${this.appConfigService.get('identityHost')}/clients`;
+    }
+
+    private getGroupsApi() {
```
Comment (DenysVuika) [inline lib/core/userinfo/services/identity-group.service.ts]: Missing return type
```suggestion
    private getGroupsApi(): string {
```

File: lib/core/userinfo/services/identity-group.service.ts [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+        return `${this.appConfigService.get('identityHost')}/groups`;
+    }
+
+    private buildRolesUrl(groupId: string): any {
```
Comment (DenysVuika) [inline lib/core/userinfo/services/identity-group.service.ts]: Missing return type
```suggestion
    private buildRolesUrl(groupId: string): string {
```

File: lib/process-services-cloud/src/lib/group/components/group-cloud.component.ts [hunk 5]
```diff
@@ -81,26 +80,26 @@ export class GroupCloudComponent implements OnInit, OnChanges {
 
     /** Emitted when a group is selected. */
     @Output()
-    selectGroup: EventEmitter<GroupModel> = new EventEmitter<GroupModel>();
+    selectGroup: EventEmitter<IdentityGroupModel> = new EventEmitter<IdentityGroupModel>();
```
Comment (DenysVuika) [inline lib/process-services-cloud/src/lib/group/components/group-cloud.component.ts]: With current TypeScript, you can simplify the assignment
```suggestion
    selectGroup = new EventEmitter<IdentityGroupModel>();
```

File: lib/process-services-cloud/src/lib/group/components/group-cloud.component.ts [hunk 6]
```diff
@@ -81,26 +80,26 @@ export class GroupCloudComponent implements OnInit, OnChanges {
...
 
     /** Emitted when a group is removed. */
     @Output()
-    removeGroup: EventEmitter<GroupModel> = new EventEmitter<GroupModel>();
+    removeGroup: EventEmitter<IdentityGroupModel> = new EventEmitter<IdentityGroupModel>();
```
Comment (DenysVuika) [inline lib/process-services-cloud/src/lib/group/components/group-cloud.component.ts]: Can be simplified
```suggestion
    removeGroup = new EventEmitter<IdentityGroupModel>();
```

File: lib/process-services-cloud/src/lib/group/components/group-cloud.component.ts [hunk 7]
```diff
@@ -81,26 +80,26 @@ export class GroupCloudComponent implements OnInit, OnChanges {
...
+    private searchGroups: IdentityGroupModel[] = [];
 
-    private searchGroupsSubject: BehaviorSubject<GroupModel[]>;
+    private searchGroupsSubject: BehaviorSubject<IdentityGroupModel[]>;
 
-    private selectedGroupsSubject: BehaviorSubject<GroupModel[]>;
+    private selectedGroupsSubject: BehaviorSubject<IdentityGroupModel[]>;
 
-    searchGroups$: Observable<GroupModel[]>;
+    searchGroups$: Observable<IdentityGroupModel[]>;
```
Comment (DenysVuika) [inline lib/process-services-cloud/src/lib/group/components/group-cloud.component.ts]: Just make this a `BehaviorSubject<IdentityGroupModel[]>` and remove the private fields. BehaviorSubject is already an Observable.

File: lib/process-services-cloud/src/lib/group/components/group-cloud.component.ts [hunk 8]
```diff
@@ -81,26 +80,26 @@ export class GroupCloudComponent implements OnInit, OnChanges {
...
 
-    selectedGroups$: Observable<GroupModel[]>;
+    selectedGroups$: Observable<IdentityGroupModel[]>;
```
Comment (DenysVuika) [inline lib/process-services-cloud/src/lib/group/components/group-cloud.component.ts]: same as above, make it a public BehaviorSubject instead

File: lib/process-services-cloud/src/lib/group/components/group-cloud.component.ts [hunk 9]
```diff
@@ -112,9 +111,9 @@ export class GroupCloudComponent implements OnInit, OnChanges {
 
     isDisabled: boolean;
 
-    constructor(private groupService: GroupCloudService) {
-        this.selectedGroupsSubject = new BehaviorSubject<GroupModel[]>(this.selectedGroups);
-        this.searchGroupsSubject = new BehaviorSubject<GroupModel[]>(this.searchGroups);
+    constructor(private identityGroupService: IdentityGroupService) {
+        this.selectedGroupsSubject = new BehaviorSubject<IdentityGroupModel[]>(this.selectedGroups);
```
Comment (DenysVuika) [inline lib/process-services-cloud/src/lib/group/components/group-cloud.component.ts]: when constructor in invoked the "selectedGroups" is always "[]", so you can initialize that in the declaration:
```
selectedGroups$ = new BehaviorSubject<IdentityGroupModel[]>([]);
```

File: lib/process-services-cloud/src/lib/group/components/group-cloud.component.ts [hunk 10]
```diff
@@ -141,7 +140,7 @@ export class GroupCloudComponent implements OnInit, OnChanges {
     }
 
     private async loadClientId() {
```
Comment (DenysVuika) [inline lib/process-services-cloud/src/lib/group/components/group-cloud.component.ts]: misses return type

File: lib/process-services-cloud/src/lib/group/components/group-cloud.component.ts (hunk 7 above)
Comment (sivakumar414ram) [reply lib/process-services-cloud/src/lib/group/components/group-cloud.component.ts]: fixed

File: lib/core/userinfo/services/identity-group.service.ts [hunk 11]
```diff
@@ -0,0 +1,347 @@
...
+     * Gets all groups.
+     * @returns Array of group information objects
+     */
+    getGroups(): Observable<IdentityGroupModel[]> {
+        const url = this.getGroupsApi();
+        const httpMethod = 'GET', pathParams = {},
+        queryParams = {}, bodyParam = {}, headerParams = {},
+        formParams = {}, authNames = [], contentTypes = ['application/json'];
+
+        return from(this.alfrescoApiService.getInstance().oauth2Auth.callCustomApi(
```
Comment (DenysVuika) [inline lib/core/userinfo/services/identity-group.service.ts]: @mauriziovitale is that ok that the API is calling the SSO directly?

Comment (mauriziovitale) [reply lib/core/userinfo/services/identity-group.service.ts]: We don't have any other choice. We need to call an API with the SSO token. In the future, the JS API will expose specific API, I guess.

Comment (DenysVuika) [reply lib/core/userinfo/services/identity-group.service.ts]: Ok, was just confirming it's expected behavior.

**Pull request merged**